### PR TITLE
tests: mutex: remove and refine some error test cases

### PR DESCRIPTION
**Pull request by enjiamai**
Remove and refine the error/negative test cases. There are three change here:

1. Remove the error test cases that trigger assertion.
2. Refine the NULL and invalid kobject parameter test case.
3. Use the common fatal error handler to reduce code.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

### CODE REVIEW

File: tests/kernel/mutex/mutex_error_case/src/test_mutex_error.c [hunk 1]
```diff
@@ -193,65 +112,106 @@ static int create_negative_test_thread(int choice)
 	return ret;
 }
 
-/* TESTPOINT: Pass a null pointer into the API k_mutex_init */
+
+/**
+ * @brief Test initializing mutex with a NULL pointer
+ *
+ * @details Pass a null pointer as a parameter then expected error happened.
```
Comment (LeiW000) [inline tests/kernel/mutex/mutex_error_case/src/test_mutex_error.c:119]: It better to change it like below.
"Pass a null pointer as parameter, then see if the expected error happens."

File: tests/kernel/mutex/mutex_error_case/src/test_mutex_error.c [hunk 2]
```diff
@@ -193,65 +112,106 @@ static int create_negative_test_thread(int choice)
...
 {
 	create_negative_test_thread(MUTEX_INIT_NULL);
 }
-/* TESTPOINT: Pass a null pointer into the API k_mutex_lock */
-static void test_mutex_lock_null(void)
+
+/**
+ * @brief Test initialize mutex with a invalid kernel object
+ *
+ * @details Pass a invalid kobject as a parameter then expected error happened.
```
Comment (LeiW000) [inline tests/kernel/mutex/mutex_error_case/src/test_mutex_error.c:133]: Same as line 119.

Same problems at several lines. Please check/correct them all.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/kernel/mutex/mutex_error_case/src/test_mutex_error.c (hunk 1 above)
Comment (enjiamai) [reply tests/kernel/mutex/mutex_error_case/src/test_mutex_error.c]: Got it fixed, thanks!

File: tests/kernel/mutex/mutex_error_case/src/test_mutex_error.c (hunk 2 above)
Comment (enjiamai) [reply tests/kernel/mutex/mutex_error_case/src/test_mutex_error.c]: Got it fixed, thanks!

**Review by LeiW000 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**